File: Find_the_Highest_Altitude.js
/**
 * @param {number[]} gain
 * @return {number}
 */
var largestAltitude = function(gain) {
    let cur = 0;
    let highest = cur;

    for (let i = 0; i < gain.length; i++) {
        let a = gain[i];
        cur += a;
        highest = Math.max(highest, cur);
    }

    return highest;    
};

File: Find_the_Highest_Altitude.java
class Solution {
    public int largestAltitude(int[] gain) {
        int cur = 0;
        int highest = cur;

        for (int i = 0; i < gain.length; i++) {
            int a = gain[i];
            cur += a;
            highest = Math.max(highest, cur);
        }

        return highest;        
    }
}
